homeassistant/components/scrape/__init__.py

```diff
@@ -1 +1,86 @@
 """The scrape component."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.rest import RESOURCE_SCHEMA, create_rest_data_from_config
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import (
+    CONF_ATTRIBUTE,
+    CONF_SCAN_INTERVAL,
+    CONF_VALUE_TEMPLATE,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.template_entity import TEMPLATE_SENSOR_BASE_SCHEMA
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_INDEX, CONF_SELECT, DEFAULT_SCAN_INTERVAL, DOMAIN
+from .coordinator import ScrapeCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SENSOR_SCHEMA = vol.Schema(
+    {
+        **TEMPLATE_SENSOR_BASE_SCHEMA.schema,
+        vol.Optional(CONF_ATTRIBUTE): cv.string,
+        vol.Optional(CONF_INDEX, default=0): cv.positive_int,
+        vol.Required(CONF_SELECT): cv.string,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+    }
+)
+
+COMBINED_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_SCAN_INTERVAL): cv.time_period,
+        **RESOURCE_SCHEMA,
+        vol.Optional(SENSOR_DOMAIN): vol.All(
+            cv.ensure_list, [vol.Schema(SENSOR_SCHEMA)]
+        ),
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {vol.Optional(DOMAIN): vol.All(cv.ensure_list, [COMBINED_SCHEMA])},
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Scrape from yaml config."""
+    if not (scrape_config := config.get(DOMAIN)):
+        return True
+
+    for resource_config in scrape_config:
+        if not (sensors := resource_config.get(SENSOR_DOMAIN)):
+            raise PlatformNotReady("No sensors configured")
```

Comment:
Raising `PlatformNotReady` only works within a platform setup function.

Reply:
Why should we raise at all? It's optional to configure the sensor's apparently so we should just continue without setting up the sensor platform.

Reply:
The exception cases obviously are missing proper tests, as they the exceptions are incorrectly implemented.

Reply:
Both issues should be fixed with #81259